corehq/apps/integration/payments/services.py

```diff
@@ -143,12 +143,14 @@ def verify_payment_cases(domain, case_ids: list, verifying_user: WebUser):
 
 def _validate_payment_cases_for_verification(case_ids, domain):
     valid_statuses = [PaymentStatus.NOT_VERIFIED, PaymentStatus.REQUEST_FAILED]
-    for case in CommCareCase.objects.iter_cases(case_ids, domain):
-        _validate_payment_case_status(case, valid_statuses, operation=_("Verification"))
+    for case in CommCareCase.objects.iter_cases(case_ids):
+        _validate_payment_case_status(case, domain, valid_statuses, operation=_("Verification"))
         _validate_final_mobile_validation(case)
 
 
-def _validate_payment_case_status(case, valid_statuses, operation):
+def _validate_payment_case_status(case, domain, valid_statuses, operation):
+    if case.domain != domain:
+        raise PaymentRequestError(_("Case '{}' not in '{}' project").format(case.case_id, domain))
```

Comment:
@ajeety4, I added a domain check here as well as a new error. Does this seem right to you?